dist/api_data.py

```diff
@@ -1208,6 +1209,37 @@ def __ge__(self, other):
         characters are hexadecimal encoded.  These formats are compatible
         with the @ref util_dump and @ref util_load commands''',
         choices=['hex', 'json', 'print']),
+    Config('incremental', '', r'''
+        configure the cursor for block incremental backup usage. These formats
+        are only compatible with the backup data source; see @ref backup''',
+        type='category', subconfig=[
+        Config('enabled', 'false', r'''
+            whether to configure this backup as an incremental starting point''',
+            type='boolean'),
+        Config('file', '', r'''
+            the file name of interest when opening a duplicate incremental backup
+            cursor. That duplicate cursor will walk the block modifications relevant
+            to the given file name'''),
+        Config('force_stop', 'false', r'''
+            causes all block incremental backup information to be released. This is
+            on an open_cursor call and the resources will be released when this
+            cursor is closed. No other operations should be done on this open cursor''',
+            type='boolean'),
+        Config('granularity', '16MB', r'''
+            this setting manages the granularity of how WiredTiger maintains modification
+            maps internally. The larger the granularity, the smaller amount of information
+            we need to maintain''',
+            min='1MB', max='2GB'),
+        Config('src_id', '', r'''
+            a string that identifies a previous checkpoint backup source as the source
+            of this incremental backup. This identifier must have already been created
+            by use of the 'this_id' configuration in an earlier backup. A source id is
+            required to begin an incremental backup'''),
+        Config('this_id', '', r'''
+            a string that identifies the most recent checkpoint as a future backup source
+            for an incremental backup via 'src_id'. This identifier is required and an
+            error will be returned if one is not provided'''),
```

Comment:
Doesn't this only need to be specified if we're doing incremental backups?

And, doesn't it identify a backup as the source, not a checkpoint?

Reply:
Yes to the first question. I'm unclear on what you mean in your second question, so some kind of rewording is probably in order. When a user opens a backup cursor with incremental and `this_id=xxx`, then we associate the current state of the system (as in most recently completed full checkpoint) as the stake in the ground for this identifier. What do you mean by "identify a backup as the source"? To me the backup is the destination not the source.

Reply:
What I'm trying to say is that we've divorced checkpoints from incremental backups -- it's incidental in the API that we're using a checkpoint somewhere, right? So, we're identifying "the backup as a future base for an incremental".